### DISCUSSION TITLE
Do we need opentelemetry-proto as separate crate?

### DISCUSSION THREAD

**Discussion by lalitb** [open-telemetry/opentelemetry-rust#1865] [General]
Just curious, do we need `opentelemetry-proto`, or can it can be integrated within `opentelemetry-otlp`. I am not sure if this crate would ever be used separately from `opentelemetry-otlp`. I remember @cijothomas also raised this earlier.

**Comment by cijothomas** [comment]
I don't think it is needed. I think of it as an internal dependency of the otlp crate only.

**Comment by TommyCpp** [comment]
Original issue https://github.com/open-telemetry/opentelemetry-rust/issues/627

This was introduced to reduce duplication between otlp and zpage(which is gone by now IIRC). There were also interest in using them in [other crates](https://crates.io/crates/opentelemetry-proto/reverse_dependencies). Also we use them in other exporters supporting OTLP protocol like [ETW exporter](https://github.com/open-telemetry/opentelemetry-rust-contrib/tree/main/opentelemetry-etw-metrics) right?

**Comment by lalitb** [comment]
Oh yes, thanks for sharing, so any exporter that needs OTLP/protobuf format would need it.
